packages/core/src/components/MonthPicker/MonthPicker.jsx

```diff
@@ -0,0 +1,220 @@
+import Button from '../Button/Button';
+import Choice from '../ChoiceList/Choice';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+/*
+`<MonthPicker>`
+
+@react-component MonthPicker
+
+Style guide: components.month-picker.react
+*/
+
+const NUM_MONTHS = 12;
+
+/**
+ * The `MonthPicker` component renders a grid of checkboxes with shortened month
+ * names as well as buttons for selecting or deselecting all. For internationalization
+ * one can pass a `locale` prop, and the month names will change to match the
+ * language of the locale. Full month names are also included as `aria-label`
+ * attributes.
+ */
+export class MonthPicker extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.months = getMonthNames(props.locale);
+    this.monthsLong = getMonthNames(props.locale, false);
+
+    if (typeof this.props.selectedMonths === 'undefined') {
+      this.isControlled = false;
+      // Since this isn't a controlled component, we need a way
+      // to track when the value has changed.
+      this.state = {
+        selectedMonths:
+          this.props.defaultSelectedMonths || this.monthsArray(false)
+      };
+    } else {
+      this.isControlled = true;
+    }
+  }
+
+  selectedMonths() {
+    if (this.isControlled) {
+      return this.props.selectedMonths;
+    }
+
+    return this.state.selectedMonths;
+  }
+
+  disabledMonths() {
+    return this.props.disabledMonths || this.monthsArray(false);
+  }
+
+  /**
+   * Generates an array of 12 booleans with the given value
+   *
+   * @param  {boolean} [selected] value to populate array items
+   * @return {boolean[]}          array of 12 booleans
+   */
+  monthsArray(selected) {
+    const array = [];
+    for (let i = 0; i < NUM_MONTHS; i++) {
+      array[i] = selected;
+    }
+    return array;
+  }
+
+  handleChange(event) {
+    if (this.props.onChange) {
+      this.props.onChange(event);
+    }
+
+    if (!this.isControlled) {
+      const index = event.target.value;
+      const selectedMonths = this.state.selectedMonths.slice();
+      selectedMonths[index] = !selectedMonths[index];
+      this.setState({ selectedMonths });
+    }
+  }
+
+  handleSelectAll() {
+    if (this.props.onSelectAll) {
+      this.props.onSelectAll();
+    }
+
+    if (!this.isControlled) {
+      this.setState({
+        selectedMonths: this.disabledMonths().map(disabled => !disabled)
+      });
+    }
+  }
+
+  handleClearAll() {
+    if (this.props.onClearAll) {
+      this.props.onClearAll();
+    }
+
+    if (!this.isControlled) {
+      this.setState({ selectedMonths: this.monthsArray(false) });
+    }
+  }
+
+  renderMonths(selectedMonths) {
+    const disabledMonths = this.disabledMonths();
+    const { name, inversed } = this.props;
+    return (
+      <ul className="ds-c-list--bare ds-u-display--flex ds-u-justify-content--between ds-u-flex-wrap--wrap">
```

Comment:
Nit: I think `ol` would be more semantic for this list since the order is meaningful.

Via [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ol):

> The `<ol>` and `<ul>` both represent a list of items. They differ in the way that, with the `<ol>` element, the order is meaningful. As a rule of thumb to determine which one to use, try changing the order of the list items; if the meaning is changed, the `<ol>` element should be used, else the `<ul>` is adequate.

Reply:
Doh

Reply:
Oh, I said "Doh" because when I first read this I thought we were talking about SubSteps. So don't make the same comment on that one until I change it.